packages/dev/core/src/Inputs/scene.inputManager.ts

```diff
@@ -201,45 +207,40 @@ export class InputManager {
             }
         }
 
-        const isMeshPicked = pickResult && pickResult.hit && pickResult.pickedMesh ? true : false;
-        if (isMeshPicked) {
-            scene.setPointerOverMesh(pickResult!.pickedMesh, evt.pointerId, pickResult);
-
-            if (!scene.doNotHandleCursors && canvas && this._pointerOverMesh) {
-                const actionManager = this._pointerOverMesh._getActionManagerForTrigger();
-                if (actionManager && actionManager.hasPointerTriggers) {
-                    canvas.style.cursor = actionManager.hoverCursor || scene.hoverCursor;
-                }
-            }
-        } else {
-            scene.setPointerOverMesh(null, evt.pointerId, pickResult);
-        }
+        this._setCursorAndPointerOverMesh(pickResult, evt.pointerId, scene);
 
         for (const step of scene._pointerMoveStage) {
+            const isMeshPicked = pickResult && pickResult.hit && pickResult.pickedMesh ? true : false;

```

Comment:
would pickResult?.pickedMesh not be sufficient ?

Reply:
Changed